File: FeedItForward/backend/app/schemas/hawker.py
from pydantic import BaseModel
from typing import Optional

from .user import User, UserCreate, UserUpdate, Role
from .misc import Geometry

class Hawker(BaseModel):
    hawker_id: int
    overall_rating: float
    business_name: str
    operating_hours: str
    food_type: str
    geometry: Geometry
    is_registered: bool

    user_id: int
    user: User

    class Config:
        orm_mode = True

class HawkerCreate(UserCreate):
    role: Role = Role.HAWKER
    business_name: str
    operating_hours: str
    food_type: str
    geometry: Geometry

class HawkerUpdate(UserUpdate):
    hawker_id: int
    business_name: str
    operating_hours: str
    food_type: str
    geometry: Geometry

File: FeedItForward/frontend/src/schemas/hawker.ts
import { Review } from "./review";
import { UserCreate, UserDisplay, UserUpdate } from "./user";
import { Geometry } from "./misc";

export interface Hawker {
  hawker_id: number;
  business_name: string;
  food_type: string;
  operating_hours: string;
  overall_rating: number;
  geometry: Geometry;
  is_registered: boolean;

  user_id: number;
  user: UserDisplay;
}

export interface HawkerCreate extends UserCreate {
  business_name: string;
  operating_hours: string;
  food_type: string;
  geometry: Geometry;
}

export interface HawkerUpdate extends UserUpdate {
  hawker_id: number;
  business_name: string;
  operating_hours: string;
  food_type: string;
  geometry: Geometry;
}
